src/adagrams.js

```diff
@@ -1,6 +1,165 @@
+        'S', 'T', 'T', 'T', 'T', 'T', 'T', 'U', 'U', 'U',
+        'U', 'V', 'V', 'W', 'W', 'X', 'Y', 'Y', 'Z'
+  ],
+  scoreChart: {
+    A: 1,
+    B: 3,
+    C: 3,
+    D: 2,
+    E: 1,
+    F: 4,
+    G: 2,
+    H: 4,
+    I: 1,
+    J: 8,
+    K: 5,
+    L: 1,
+    M: 3,
+    N: 1,
+    O: 1,
+    P: 3,
+    Q: 10,
+    R: 1, 
+    S: 1,
+    T: 1,
+    U: 1,
+    V: 4,
+    W: 4,
+    X: 8,
+    Y: 4,
+    Z: 10
+  },
+
+  shufflePool() {
+    let poolCopy = this.pool.slice(0, this.pool.length);
+    let newPosition = null;
+    let temp = null;
+
+    for (let i = poolCopy.length - 1; i > 0; i--) {
+      newPosition = Math.floor(Math.random() * (i + 1));
+
+      temp = poolCopy[i];
+      poolCopy[i] = poolCopy[newPosition];
+      poolCopy[newPosition] = temp;
+    }
+
+    return poolCopy;
+  },
+
   drawLetters() {
-    // Implement this method for wave 1
+    let shuffledPool = Adagrams.shufflePool();
+
+    let drawn = shuffledPool.slice(0, 10);
+
+    return drawn;
+  },
+
+  usesAvailableLetters(input, lettersInHand) { 
+    let lettersCount = {};
+
+    lettersInHand.forEach((letter) => {
+      if (!lettersCount[letter]) {
+        lettersCount[letter] = 1; 
+      } else {
+        lettersCount[letter] += 1;
+      }
+    });
+
+    let wordArray = input.toUpperCase().split('');
+
+    for(let i = 0; i < wordArray.length; i++) {
+      const letter = wordArray[i];
+      if(lettersCount[letter] === undefined || lettersCount[letter] === 0) {
+        return false;
+      } else {
+        lettersCount[letter] -= 1;
+      }
+    };
+
+    return true;
+  },
+
+  scoreWord(word) {
+    let score = 0;
+    if (word.length > 6) {
+      score += 8;
+    }
+
+    word.toUpperCase().split('').forEach((letter) => {
+      score += this.scoreChart[letter];
+    })
+
+    return score;
+  },
+
+  scoreWords(words) {
+    const scoreOfWords = {};
+
+    words.forEach((word) => scoreOfWords[word] = Adagrams.scoreWord(word));
+
+    return scoreOfWords;
+  },
+
+  highestScoreFrom(words) {
+    const scoredWords = Adagrams.scoreWords(words);
+
+    let max = 0;
+    for(const value of Object.values(scoredWords)) {
+      if (value > max) {
+        max = value;
+      }
+    }
+
+    let maxScoredWords = [];
+    // for of: loops over values generated by Object.entries or (Object.keys or Object.values)
+    // Object.entries(scoredWords) returns both key and value
+    // That's why we need [key, value]. Should be with square brackets
+    for(const [key, value] of Object.entries(scoredWords)) {
+      if(value === max) {
+        maxScoredWords.push(key);
+      }
+    }
+    
+    if (maxScoredWords.length === 1) {
+      let word = maxScoredWords[0]
+      return { score: max, word: word };
+    } else {
+      return Adagrams.tieBreakingLogic(maxScoredWords, max);
+    }
+  },
+
+  tieBreakingLogic(maxScoredWords, maxScore) {
+    let minLength = maxScoredWords[0].length;
+
+    for(let i = 0; i < maxScoredWords.length; i++) {
```

Comment:
Is there a way that you can refactor this so that you don't have two separate for loops and you're available to chain together some of the if statements into if / else if?